osu.Game/Overlays/Home/Friends/FriendsOnlineStatusItem.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.Home.Friends
+{
+    public class FriendsOnlineStatusItem : OverlayUpdateStreamItem<FriendsBundle>
+    {
+        public FriendsOnlineStatusItem(FriendsBundle value)
+            : base(value)
+        {
+        }
+
+        protected override string GetMainText => Value.Status.ToString();
+
+        protected override string GetAdditionalText => Value.Count.ToString();
+
+        protected override Color4 GetBarColour(OsuColour colours)
+        {
+            switch (Value.Status)
+            {
+                default:
+                    throw new ArgumentException($@"{Value.Status} status does not provide a colour in {nameof(GetBarColour)}.");
```

Comment:
I actually had to go and check that C# does the intelligent thing and checks all of the case labels beforehand instead of doing a glorified `goto` like C does. Please move this case to last, it reads really confusing otherwise.